### PR TITLE
fix: LUP scripts and wandering vendor script

### PR DESCRIPTION
**Pull request by EmosewaMC**
Just implements the scripts as-is from the moreMovementAI branch so I can work that branch down to be more workable.

tested that red green and yellow bots waved when interacted with tested that construction robot races when interacted with wandering vendor does nothing before and after, but script is ready for use when npcs are implemented.

### CODE REVIEW

**Review by aronwk-aaron — APPROVED** [review]

File: dScripts/CppScripts.cpp [hunk 1]
```diff
@@ -547,7 +550,7 @@ CppScripts::Script* CppScripts::GetScript(Entity* parent, const std::string& scr
 	//PR:
 	else if (scriptName == "scripts\\client\\ai\\PR\\L_PR_WHISTLE.lua")
 		script = new PrWhistle();
-	else if (scriptName == "scripts\\02_server\\Map\\PR\\L_PR_SEAGULL_FLY.lua")
+	if (scriptName == "scripts\\02_server\\Map\\PR\\L_PR_SEAGULL_FLY.lua")
```
Comment (jadebenn) [inline dScripts/CppScripts.cpp:553]: Was the deletion of else here intentional?

File: dScripts/CppScripts.cpp [hunk 2]
```diff
@@ -726,7 +731,7 @@ CppScripts::Script* CppScripts::GetScript(Entity* parent, const std::string& scr
 		script = new NTNaomiDirtServer();
 
 	//AM:
-	else if (scriptName == "scripts\\02_server\\Map\\AM\\L_AM_CONSOLE_TELEPORT_SERVER.lua")
```
Comment (jadebenn) [inline dScripts/CppScripts.cpp:729]: Ditto

File: dScripts/CppScripts.cpp [hunk 3]
```diff
@@ -806,7 +811,7 @@ CppScripts::Script* CppScripts::GetScript(Entity* parent, const std::string& scr
 		script = new Lieutenant();
 	else if (scriptName == "scripts\\02_server\\Map\\njhub\\L_RAIN_OF_ARROWS.lua")
 		script = new RainOfArrows();
-	else if (scriptName == "scripts\\02_server\\Map\\njhub\\L_CAVE_PRISON_CAGE.lua")
```
Comment (jadebenn) [inline dScripts/CppScripts.cpp:809]: And here

**Review by jadebenn — CHANGES_REQUESTED** [review]

**Comment by jadebenn** [comment]
Confirmed on Discord that the dropped elses are intentional due to win32 reasons.

**Comment by aronwk-aaron** [comment]
Win32 (futher explained on discord)

**Review by jadebenn — APPROVED** [review]

**Pull request merged**